File: timetrials/src/main/java/fzero/timetrials/SubmissionApplicationService.java
package fzero.timetrials;

import fzero.timetrials.dto.SubmissionDto;
import fzero.timetrials.game.TimeTrialRun;
import fzero.timetrials.repository.TimetrialRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class SubmissionApplicationService {
    @Autowired
    private TimetrialRepository timetrialRepository;

    public void saveRun(SubmissionDto submissionDto) {
        TimeTrialRun timeTrialRun = new TimeTrialRun();
        timeTrialRun.setMap(submissionDto.getMap());
        timeTrialRun.setTotalTime(submissionDto.getTotalTime());
        timeTrialRun.setRound1(submissionDto.getRound1());
        timeTrialRun.setRound2(submissionDto.getRound2());
        timeTrialRun.setRound3(submissionDto.getRound3());
        timeTrialRun.setRound4(submissionDto.getRound4());

        timetrialRepository.save(timeTrialRun);
    }
}

File: FZero99-TT/src/main/kotlin/fZero/fZero99TT/applicationService/SubmissionApplicationService.kt
package fZero.fZero99TT.applicationService

import fZero.fZero99TT.dto.SubmissionDto
import fZero.fZero99TT.game.TimeTrialRun
import fZero.fZero99TT.repository.TimetrialRepository
import org.springframework.stereotype.Service

@Service
class SubmissionApplicationService(
        private val timetrialRepository: TimetrialRepository,
) {
    fun saveRun(submissionDto: SubmissionDto) {
        timetrialRepository.save(
            TimeTrialRun(
                track = submissionDto.track,
                totalTime = submissionDto.totalTime,
                round1 = submissionDto.round1,
                round2 = submissionDto.round2,
                round3 = submissionDto.round3,
                round4 = submissionDto.round4,
        ))
    }
}
